lrauv_ignition_plugins/test/test_sensor.cc

```diff
@@ -99,58 +152,44 @@ TEST(SensorTest, Sensor)
-  ignition::math::Angle lat1 = IGN_DTOR(36.7999992370605);
-  ignition::math::Angle lon1 = IGN_DTOR(-122.720001220703);
-  {
-    lrauv_ignition_plugins::msgs::LRAUVInit spawnMsg;
-    spawnMsg.mutable_id_()->set_data("vehicle1");
-    spawnMsg.set_initlat_(lat1.Degree());
-    spawnMsg.set_initlon_(lon1.Degree());
-    spawnMsg.set_initz_(spawnDepth);
-    spawnMsg.set_acommsaddress_(201);
-
-    spawnPub.Publish(spawnMsg);
-  }
+  // On a depth coplanar to a piece of data
+  SpawnVehicle(
+    spawnPub, "vehicle1", 36.7999992370605, -122.720001220703, 50, 0);
+  node.Subscribe("/model/vehicle1/chlorophyll", &ChlorophyllVeh1Cb);
+
+  // At the surface should still have reading
+  SpawnVehicle(
+    spawnPub, "vehicle2", 36.7999992370605, -122.720001220703, 0, 0);
+  node.Subscribe("/model/vehicle2/chlorophyll", &ChlorophyllVeh2Cb);
+
+  // In the middle of two slices, still in a plane
+  SpawnVehicle(
+    spawnPub, "vehicle3", 36.7999992370605, -122.720001220703, 45, 0);
+  node.Subscribe("/model/vehicle3/chlorophyll", &ChlorophyllVeh3Cb);
+
+  // In the middle of a quadrant
+  SpawnVehicle(
+    spawnPub, "vehicle4", 36.7999992370605, -122.7, 48, 0);
+  node.Subscribe("/model/vehicle4/chlorophyll", &ChlorophyllVeh4Cb);
 
   // Check that vehicle was spawned
   int expectedIterations = iterations;
-  for (sleep = 0; ignition::gazebo::kNullEntity == vehicle1 && sleep < maxSleep;
+  for (sleep = 0; spawnedAllVehicles.load() == false && sleep < maxSleep;
       ++sleep)
   {
     std::this_thread::sleep_for(100ms);
     // Run paused so we avoid the physics moving the vehicles
     fixture->Server()->RunOnce(true);
     expectedIterations++;
-    igndbg << "Waiting for vehicle1" << std::endl;
+    igndbg << "Waiting for vehicles to spawn" << std::endl;
   }
-  EXPECT_LE(sleep, maxSleep);
-  EXPECT_NE(ignition::gazebo::kNullEntity, vehicle1);
-  EXPECT_EQ(expectedIterations, iterations);
-  EXPECT_LT(0, poses1.size());
-  EXPECT_LT(0, latLon1.size());
-
-  double tightTol{1e-5};
-
-  // Check vehicle positions in ENU
-  // World origin
-  EXPECT_NEAR(0.0, poses1.back().Pos().X(), tightTol);
-  EXPECT_NEAR(0.0, poses1.back().Pos().Y(), tightTol);
-  EXPECT_NEAR(0.0, poses1.back().Pos().Z(), tightTol);
-  // Facing North (-90 rotation from default West orientation)
-  EXPECT_NEAR(0.0, poses1.back().Rot().Roll(), tightTol);
-  EXPECT_NEAR(0.0, poses1.back().Rot().Pitch(), tightTol);
-  EXPECT_NEAR(-IGN_PI*0.5, poses1.back().Rot().Yaw(), tightTol);
+  EXPECT_TRUE(spawnedAllVehicles.load());
 
-  EXPECT_NEAR(lat1.Degree(), latLon1.back().X(), tightTol);
-  EXPECT_NEAR(lon1.Degree(), latLon1.back().Y(), tightTol);
-  EXPECT_NEAR(-spawnDepth, latLon1.back().Z(), tightTol);
-
-  // Higher tolerance for lat/lon because of the conversions
-  double latLonTol{2e-2};
-
-  fixture->Server()->Run(true, 10000, false);
+  fixture->Server()->Run(true, 10, false);
 
+  for (auto msg: received_msg)
+  {
+    EXPECT_TRUE(msg);
```

Comment:
@arjo129 meta: this is racing with Ignition Transport. Consider using a timeout (measured against steady or system clocks) instead. 

Writing this kind of tests will get easier once #195 lands.

Reply:
Good point. I think we could wait till #195 goes in before I rewrite this.